logstash-core/src/main/java/org/logstash/ext/JrubyAckedWriteClientExt.java

```diff
@@ -50,16 +56,15 @@ private JrubyAckedWriteClientExt(final Ruby runtime, final RubyClass metaClass,
     @JRubyMethod(name = {"push", "<<"}, required = 1)
     public IRubyObject rubyPush(final ThreadContext context, IRubyObject event) {
         ensureOpen();
-        queue.rubyWrite(context, ((JrubyEventExtLibrary.RubyEvent) event).getEvent());
+        queue.rubyWrite(context, Collections.singletonList(((JrubyEventExtLibrary.RubyEvent) event).getEvent()), -1);
         return this;
     }
 
     @JRubyMethod(name = "push_batch", required = 1)
     public IRubyObject rubyPushBatch(final ThreadContext context, IRubyObject batch) {
         ensureOpen();
-        for (final IRubyObject event : (Collection<JrubyEventExtLibrary.RubyEvent>) batch) {
-            queue.rubyWrite(context, ((JrubyEventExtLibrary.RubyEvent) event).getEvent());
-        }
+        List<Queueable> events = ((List<JrubyEventExtLibrary.RubyEvent>) batch).stream().map(e -> e.getEvent()).collect(Collectors.toList());
```

Comment:
@andrewvc this is pretty expensive. I think going a little more old school here and simply creating a new ArrayList(batch.size()) and then filling it in a for loop fill be quite a bit faster.
On that note ... why not simply create the List<byte[]> here and adjust the downstream APIs. Unlike the name suggests those aren't used by Ruby anymore so you're safe to change them and save one List and conversion step :)